### PR TITLE
Include XKit version as a header in requests to Tumblr

### PR DESCRIPTION
**Pull request by cyle**
This updates the `XKit.tools.Nx_XHR` function and the `GM_xmlhttpRequest` functions to include a new `X-XKit-Version` header which could allow Tumblr to identify requests being made via XKit, instead of them looking like any other AJAX request on the site.

Did I catch all the places to do this? 😄 

@nightpool @AprilSylph

### CODE REVIEW

**Comment by AprilSylph** [comment]
Nx_XHR needs to be updated in `Extensions/xkit_patches.js` under the 7.9.0 function, which will be shuffled into `xkit.js` prior to a 7.9.1 release (since we're touching actual addon files here)

**Comment by cyle** [comment]
@AprilSylph i believe that's what i'm doing here: https://github.com/new-xkit/XKit/pull/1744/files#diff-f3c602f5caa159014c2f7557700de10cR393 or did i miss a spot?

**Comment by AprilSylph** [comment]
either i missed that or got confused since you also made a change to it in `xkit.js` that's gonna get overwritten. that's fine though

**Comment by cyle** [comment]
like that https://github.com/new-xkit/XKit/pull/1744/commits/0b428ad9091167f85bf368f3c14607ab988f20f0 @AprilSylph ?

**Comment by AprilSylph** [comment]
yup, that's all!

File: Extensions/xkit_patches.js [hunk 1]
```diff
@@ -166,6 +166,14 @@ XKit.extensions.xkit_patches = new Object({
 	patches: {
 		"7.9.0": function() {
 
+			let global = globalThis || window;
```
Comment (AprilSylph) [inline Extensions/xkit_patches.js]: can't use `globalThis` on Firefox 64 and below, which is important considering our userbase is generally update-averse - need to support down to Firefox 48 ideally or any pre-Quantum version if not

it'd also break Opera fwiw

image

**Comment by cyle** [comment]
thanks for the help folks!

**Pull request merged**